src/Metalama.Patterns.NotifyPropertyChanged/Implementation/Natural/NaturalAspect.Templates.cs

```diff
@@ -0,0 +1,713 @@
+﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
+
+using Metalama.Framework.Aspects;
+using Metalama.Framework.Code;
+using Metalama.Framework.Code.Invokers;
+using Metalama.Framework.Code.SyntaxBuilders;
+using System.ComponentModel;
+
+namespace Metalama.Patterns.NotifyPropertyChanged.Implementation.Natural;
+
+internal partial class NaturalAspect
+{
+    [CompileTime]
+    private static void CompileTimeThrow( Exception e ) => throw e;
+
+    [CompileTime]
+    private static T ExpectNotNull<T>( T? obj ) => obj ?? throw new InvalidOperationException( "A null value was not expected here." );
+
+    // ReSharper disable once EventNeverSubscribedTo.Global
+    [InterfaceMember]
+    public event PropertyChangedEventHandler? PropertyChanged;
+
+    [Template]
+    private static dynamic? OverrideInpcRefTypeProperty
+    {
+        set
+        {
+            var ctx = (BuildAspectContext) meta.Tags["ctx"]!;
```

Comment:
It would be more readable to use an accessor setter, which can have compile-time parameters without resorting to tags.